src/commands/dev/lib/builder-cache.ts

```diff
@@ -0,0 +1,150 @@
+import chalk from 'chalk';
+import execa from 'execa';
+import { join } from 'path';
+import npa from 'npm-package-arg';
+import mkdirp from 'mkdirp-promise';
+import { funCacheDir } from '@zeit/fun';
+import { readJSON, writeJSON, remove } from 'fs-extra';
+import cacheDirectory from 'cache-or-tmp-directory';
+import wait from '../../../util/output/wait';
+import { Output } from '../../../util/output';
+import { NowError } from '../../../util/now-error';
+import { devDependencies as nowCliDeps } from '../../../../package.json';
+import { Builder } from './types';
+
+import * as staticBuilder from './static-builder';
+
+const localBuilders: { [key: string]: Builder } = {
+  '@now/static': staticBuilder
+};
+
+export const cacheDirPromise = prepareCacheDir();
+export const builderDirPromise = prepareBuilderDir();
+
+/**
+ * Prepare cache directory for installing now-builders
+ */
+export async function prepareCacheDir() {
+  const designated = cacheDirectory('co.zeit.now');
+
+  if (!designated) {
+    throw new NowError({
+      code: 'NO_BUILDER_CACHE_DIR',
+      message: 'Could not find cache directory for now-builders.',
+      meta: {}
+    });
```

Comment:
Errors are usually added to https://github.com/zeit/now-cli/blob/canary/src/util/errors-ts.ts